tests/lib/rules/render-result-naming-convention.test.ts

```diff
@@ -0,0 +1,306 @@
+        import { render } from '@testing-library/react';
+        
+        const setup = () => render(<SomeComponent />);
+        
+        test('should not report render result called "view" from wrapping function', async () => {
+          const view = setup();
+          await view.findByRole('button');
+        });
+      `,
+    },
+    {
+      code: `
+        import { customRender } from 'test-utils';
+        
+        test('should not report straight destructured render result from custom render', () => {
+          const { rerender, getByText } = customRender(<SomeComponent />);
+          const button = getByText('some button');
+        });
+      `,
+      options: [
+        {
+          renderFunctions: ['customRender'],
+        },
+      ],
+    },
+    {
+      code: `
+        import { customRender } from 'test-utils';
+        
+        test('should not report render result called "view" from custom render', async () => {
+          const view = customRender();
+          await view.findByRole('button');
+        });
+      `,
+      options: [
+        {
+          renderFunctions: ['customRender'],
+        },
+      ],
+    },
+    {
+      code: `
+        import { customRender } from 'test-utils';
+        
+        test('should not report render result called "utils" from custom render', async () => {
+          const utils = customRender();
+          await utils.findByRole('button');
+        });
+      `,
+      options: [
+        {
+          renderFunctions: ['customRender'],
+        },
+      ],
+    },
+    {
+      code: `
+        import { render } from '@foo/bar';
+        
+        test('should not report from render not related to testing library', () => {
+          const wrapper = render(<SomeComponent />);
+          const button = wrapper.getByText('some button');
+        });
+      `,
+    },
+    {
+      code: `
+        import { render } from '@foo/bar';
+        
+        test('should not report from render not imported from testing library', () => {
+          const wrapper = render(<SomeComponent />);
+          const button = wrapper.getByText('some button');
+        });
+      `,
+    },
+    {
+      code: `
+        function render() {
+          return 'whatever';
+        }
+        
+        test('should not report from custom render not related to testing library', () => {
+          const wrapper = render(<SomeComponent />);
+          const button = wrapper.getByText('some button');
+        });
+      `,
+    },
+    {
+      code: `
+        import { render } from '@testing-library/react';
+        
+        const setup = () => {
+          // this one must have a valid name
+          const view = render(<SomeComponent />);
+          return view;
+        };
+
+        test('should not report render result called "view" from wrapping function', async () => {
+          // this isn't a render technically so it can be called "wrapper"
+          const wrapper = setup();
```

Comment:
I'm not sure if this should be valid or not. Should the rule just check whatever comes directly from `render`? Should check also return value from functions wrapping `render` too?

Reply:
Ideally, If `setup` was in another file, it would qualify as a `customRender` function (if configured)- is that correct? We should probably apply the same criteria.

However, in a more realistic way (?), I am not sure if you can easily ensure that whatever was returned from render is returned to the outside in this function where `render` is called. Consider the following scenario

```js
const commonLogic = async (param) => {
   // this is fine according to the rule
   const view = render(<Foo prop1={param} />
   await userEvent.click(view.getByRole('button')
   // might be more logic here
   return view.getByTestId('box').textContent
}

it('something1', () => {
  // this should be fine - a render is inside and we don't return anything render related
  const result = commonLogic(param1)
  expect(result).toBe('foo') 
})

it('something2', () => {
  // same here
  const result = commonLogic(param2)
  expect(result).toBe('baz') 
})

```

here, `commonTest` is a function that has a `render` but it does not wrap it, it just avoids duplication of some logic.

Can we differentiate this scenario from the one in your test?

Reply:
> Ideally, If setup was in another file, it would qualify as a customRender function (if configured)- is that correct? 

That's correct. It would also qualify as a render function to be checked if added to `renderFunctions` rule args (which is what I do on my projects as I tend to have tons of `setup` inline wrappers).

I'm worried about exactly that example you wrote. I don't see an easy way to spot that, and I'm not sure if should be reported either. Maybe a nice to have for a future version?

Reply:
> Maybe a nice to have for a future version?

yeah, I agree 👍 . For now, being careful of not causing false positives in these scenarios should be enough